rmw_fastrtps_cpp/src/rmw_publisher.cpp

```diff
@@ -34,6 +34,27 @@ using TopicDataType = eprosima::fastrtps::TopicDataType;
 
 extern "C"
 {
+rmw_ret_t
+rmw_init_publisher_allocation(
+  const rosidl_message_type_support_t * type_support,
+  const rosidl_message_bounds_t * message_bounds,
+  rmw_publisher_allocation_t * allocation)
+{
+  // Unused in current implementation.
+  (void) type_support;
+  (void) message_bounds;
+  (void) allocation;
+  return RMW_RET_OK;
```

Comment:
I know that this is a stub that is meant to be filled out later.  But until later comes, I think this should probably return an error since we aren't actually doing what the user asked.  Same below.  Thoughts?